### PR TITLE
[PRODDEV-219] Implement redirect after login

### PR DESCRIPTION
**Pull request by NightHunterSV**
**Related Issue/Ticket:**
https://openy.atlassian.net/browse/PRODDEV-219

## Steps to test:

- [ ] Visit site as an anonymous
- [ ] Navigate through the site using the main navigation, for example
image

- [ ] You should be redirected to the login page
image

- [ ] Log in
- [ ] Ensure you've been redirected to the needed location after the login
image

- [ ] Change the auth provider and ensure it is also working

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [x] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

File: js/gated-content/src/main.js [hunk 1]
```diff
@@ -45,6 +48,12 @@ new Vue({
   mounted() {
     const app = this;
 
+    const cookieName = 'Drupal_visitor_gc_auth_destination';
```
Comment (hamrant) [inline js/gated-content/src/main.js:51]: Let's rename `Drupal_visitor_gc_auth_destination` to `openy_gc_auth_destination`

File: modules/openy_gc_auth/js/auth.behaviors.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ Drupal.behaviors.openy_gc_auth_store_hash = {
+   attach: function (context) {
+     // Only run this script on full documents, not ajax requests.
+     if (context !== document) {
+       return;
+     }
+     if (!window.location.hash || window.location.hash === '#') {
+       return;
+     }
+     $.cookie('Drupal_visitor_gc_auth_destination', window.location.hash);
```
Comment (hamrant) [inline modules/openy_gc_auth/js/auth.behaviors.js:11]: Let's rename `Drupal_visitor_gc_auth_destination` to `openy_gc_auth_destination`

File: modules/openy_gc_auth/openy_gc_auth.libraries.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+auth:
```
Comment (hamrant) [inline modules/openy_gc_auth/openy_gc_auth.libraries.yml:1]: Let's rename library name `auth` to `auth_destination`

File: modules/openy_gc_auth/src/Plugin/Block/VirtualYLoginBlock.php [hunk 4]
```diff
@@ -103,6 +102,7 @@ public function build() {
     }
     // phpcs:enable
 
+    $form['#attached']['library'][] = 'openy_gc_auth/auth';
```
Comment (hamrant) [inline modules/openy_gc_auth/src/Plugin/Block/VirtualYLoginBlock.php:105]: Let's rename library name `auth` to `auth_destination`

**Review by hamrant — CHANGES_REQUESTED** [review]
@NightHunterSV let's rename some things in PR

**Review by hamrant — APPROVED** [review]

**Comment by AnastasiiaPys** [comment]
Tested for dummy, reclique, y-usa providers. Testing passed.

**Review by anpolimus — APPROVED** [review]

**Pull request merged**